src/here-studio.ts

```diff
@@ -0,0 +1,298 @@
+
+        //Check if the layer has tags eg. Building Footprints tags -> In Such cases copy the whole currentLayer as is -> Otherwise copy the space from the user
+        if (currentLayer.meta
+            && currentLayer.meta.tags
+            && currentLayer.meta.tags.length > 0) {
+
+            //Update the project layer as is without any modifications
+            updatedLayersData.push(currentLayer)
+        }
+        else {
+
+            //Download the space locally and reference the space for current user
+            let geoSpaceIDToCopy = currentLayer.geospace.id;
+            console.log("\nCopying layer ["+(i+1)+"] : '"+geoSpaceIDToCopy+"' from published project")
+
+            //Download the GeospaceID from published project with the publisher's token -  Download the space from GET Search and save it in local temp file - https://xyz.api.here.com/hub/spaces/uLqEizJW/search
+            let geoSpaceDownloadOptions = {
+                token : publishersToken
+            }
+            let geoSpaceData = await getSpaceDataFromXyz(geoSpaceIDToCopy, geoSpaceDownloadOptions);
+            if (geoSpaceData.features && geoSpaceData.features.length === 0) {
+                console.log("\nNo features are available to download");
+                //process.exit();
+            }
+
+            //Check if GeoSpaceData is blank -> if yes move on to next one else create the file with that name
+            let geoSpaceFileName = geoSpaceIDToCopy+".geojson";
+            await fs.writeFileSync(geoSpaceFileName, JSON.stringify(geoSpaceData));
+            console.log("Space '"+geoSpaceIDToCopy+"' downloaded locally")
+
+            //Copy the contents of the downloaded space to currentUser's XYZ spaces
+            let newSpaceData = await createSpace ({})//createNewSpaceAndUpdateMetadata(''+geoSpaceIDToCopy, ""+geoSpaceIDToCopy, {});
+            let currentGeoSpaceID = newSpaceData.id;
+
+            //Update the geospace id for current layer
+            currentLayer.geospace.id = currentGeoSpaceID;
+
+            let uploadOptions = {
+                title: geoSpaceFileName,
+                description: "GeoSpace created from HERE CLI",
+                file: geoSpaceFileName,
+                stream: true
+            }
+
+            //Upload it to current user's space
+            await uploadToXyzSpace (currentGeoSpaceID, uploadOptions);
+
+            //Clear cache and delete the file from temp repo
+            await fs.unlinkSync(geoSpaceFileName)
+
+            //Update the modified layer data.
+            updatedLayersData.push(currentLayer)
+        }
+    }
+
+    //Create a new project under current user with the settings of published projects
+    uri = "/project-api/projects";
+
+    //Update the layer data to cloned project - updatedLayersData
+    clonedProjectData.layers = updatedLayersData;
+
+    //Update the token for current project
+    clonedProjectData.rot = currentUsersToken;
```

Comment:
we can not use this token as this token has full access which should not be shared in projects data

Reply:
Updated - creating a readOnlyToken for currentUsersToken and supplying that to cloned projectsData

Reply:
I dont see changes for this

Reply:
Reference from line 121 above where we are getting a readOnly token 

```
    //Fetch current user's read only token
    let currentUsersToken = await common.verify(true);
```